extensions/azurecore/src/azureResource/utils.ts

```diff
@@ -20,14 +20,33 @@ import * as Constants from '../constants';
 import { getProxyEnabledHttpClient } from '../utils';
 import { HttpClient } from '../account-provider/auths/httpClient';
 import { NetworkRequestOptions } from '@azure/msal-common';
+import { ErrorResponseBody } from '@azure/arm-subscriptions/esm/models';
 
 const localize = nls.loadMessageBundle();
 
+/**
+ * Specialized version of the ErrorResponseBody that is required to have the error
+ * information for easier typing support, without it how do we know it's an error
+ * response?
+ * https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/subscription/arm-subscriptions/src/models/index.ts#L180
```

Comment:
We may want to also investigate this error response body stuff...the typings we have are pretty out of date and the new one has a slightly different format with some properties possibly being at the top level (instead of the error property). 

@cheenamalhotra @cssuh I'll leave it up to you what we do here, I can open an issue with details if you'd like though.

Reply:
I see we're using v3.0.0 of [@azure/arm-subscriptions](https://www.npmjs.com/package/@azure/arm-subscriptions?activeTab=versions), possibly that's why.